### PR TITLE
Change how ovsrpro projects are found downstream

### PR DESCRIPTION
**Pull request by brobeson**
- I removed the use-op CMake files.
- I added CMake Find modules for each project, except Qt5. These modules follow canonical CMake practices, and create namespaced targets such as `librdkafka::librdkafka`.
- I added a couple files to facilitate working on ovsrpro within VS Code.
- I removed opfunmac.cmake and opopts.cmake. They are not used anymore.
- I added a warning if someone tries to build ovsrpro with an unsupported compiler.
- I simplified Findovsrpro.cmake.
- I updated the patch for HDF5. I could not find a build-time method to stop the HDF5 build system from generating FindHDF5.cmake. I was to use ours, which adds targets. I had to create a patch to remove the HDF5 CMake code that generates the file.
- I restored librdkafka to our current version to minimize impacts to a downstream project. I'll revisit upgrading this later.
- I removed Qt5 patches that are not used.
- I downgraded Qt5 from 5.14 to 5.12. 5.14 and 5.13 *may* have caused build errors downstream.
- I updated all the documentation in readme.md.

### CODE REVIEW

File: Findovsrpro.cmake [hunk 1]
```diff
@@ -1,132 +1,111 @@
...
+  )
+endif()
+
+# Since ovsrpro uses date versioning instead of semantic versioning, we cannot
+# deduce compatibility from the version. So force an exact match to be found.
+if(NOT ovsrpro_FIND_VERSION_EXACT)
+  set(ovsrpro_FIND_VERSION_EXACT true)
+  ovsrpro_message(
+    AUTHOR_WARNING
+    "Ovsrpro requires an exact version match. We recomment adding EXACT to find_package()."
```
Comment (bnhhughes) [inline Findovsrpro.cmake]: Spelling - `recomment` to `recommend`

File: Findovsrpro.cmake [hunk 2]
```diff
@@ -1,132 +1,111 @@
...
+  )
+  string(
+    REGEX REPLACE
+    "([0-9]+)\\.([0-9]+)\\.([0-9]+)?"
+    "gcc\\1\\2\\3"
+    compiler
+    ${gcc_version}
+  )
+else()
+  message(FATAL_ERROR "${CMAKE_CXX_COMPILER_ID} is not supported.")
```
Comment (bnhhughes) [inline Findovsrpro.cmake:47]: Not allowing Windows builds now?  I know we only ever kindof supported them, not attempting to support them any longer?

**Comment by bnhhughes** [comment]
I started reviewing but didn't get far before I have to run to a meeting, I'll continue later this afternoon.

File: Findovsrpro.cmake (hunk 2 above)
Comment (brobeson) [reply Findovsrpro.cmake:47]: Not until some downstream project needs Windows support.

File: cmake/Findlibzmq.cmake [hunk 3]
```diff
@@ -0,0 +1,97 @@
+#[========================================================================[.rst:
+Findlibzmq
```
Comment (bnhhughes) [inline cmake/Findlibzmq.cmake:2]: Any reason this isn't combined with the cppzmq library above?  Seems like with the other libraries the C and C++ versions were in the same Find<whatever>.cmake file.

File: readme.md [hunk 4]
```diff
@@ -1,87 +1,130 @@
...
 1. Download the SHA256 checksum file for your release.
 1. Validate the checksum.
+   ```bash
+   sha256sum --check ovsrpro-18.10.1-gcc631-64-Linux.sh.sha256
+   ```
 1. Run the installer.
+   ```bash
+   sudo ./ovsrpro-18.10.1-gcc631-64-Linux.sh --prefix=/opt/extern/ --include-subdir
+   ```
+   The recommended location for installation is */opt/ovsrpro/*. We also
```
Comment (bnhhughes) [inline readme.md:20]: If this is the recommended location, probably should change the `--prefix=/opt/extern/` in the example above to be `/opt/ovsrpro/`

File: readme.md [hunk 5]
```diff
@@ -1,87 +1,130 @@
...
+If ovsrpro is found, the following variables are set:
+
+<!-- markdownlint-disable MD013 -->
+| Variable | Description |
+|:---|:---|
+| `ovsrpro_FOUND` | This is set to `true` if ovsrpro is found, and `false` if it not. |
+| `ovsrpro_ROOT_DIR` | This is set to the root path where ovsrpro is installed. An example is */opt/ovsrpro/ovsrpro-20.04.1-gcc921-64-Linux/* |
+| `ovsrpro_INCLUDE_DIR` | This is the root path to the project headers files. Each project will have a separate subdirectory within this path. An example is */opt/ovsrpro/ovsrpro-20.04.1-gcc921-64-Linux/include/*. Ovsrpro users typically should not need to use this. |
+| `ovsrpro_LIBRARY_DIR` | This is the path to the project library files. An example is */opt/ovsrpro/ovsrpro-20.04.1-gcc921-64-Linux/lib/*. Ovsrpro users typically should not need to use this. |
+| `ovsrpro_CONFIG_PATH` | This the path to find the Qt5 CMake find scripts. This is needed to find Qt5 using `find_package()`. |
```
Comment (bnhhughes) [inline readme.md]: Spelling - `This the path` --> `This is the path`

File: readme.md [hunk 6]
```diff
@@ -1,87 +1,130 @@
...
+<!-- markdownlint-disable MD013 -->
+
+`${ovsrpro_ROOT_DIR}`*/share/cmake* is appended to `CMAKE_MODULE_PATH`
+automatically. To find ovsrpro projects, except for Qt5, add `find_package()`
+commands. The project names must all be prefixed with "Ovsrpro".
```
Comment (bnhhughes) [inline readme.md]: Is this last line still true? I didn't see where "Ovsrpro" was required when finding the different projects, and it also isn't used in the examples below.

File: readme.md [hunk 7]
```diff
@@ -1,87 +1,130 @@
...
-   https://github.com/distributePro/ovsrpro/releases.
+   <https://github.com/distributePro/ovsrpro/releases>.
 1. Download the SHA256 checksum file for your release.
 1. Validate the checksum.
+   ```bash
+   sha256sum --check ovsrpro-18.10.1-gcc631-64-Linux.sh.sha256
+   ```
 1. Run the installer.
+   ```bash
+   sudo ./ovsrpro-18.10.1-gcc631-64-Linux.sh --prefix=/opt/extern/ --include-subdir
```
Comment (bnhhughes) [inline readme.md]: I noticed elsewhere the ovsrpro version was moved to `20.04.1`, should this example (and the one above where the checksum is validated) have the newer version as well?

**Review by bnhhughes — APPROVED** [review]
That's a lot of cleanup, just a few minor things I noticed.  Looks good!

File: cmake/Findlibzmq.cmake (hunk 3 above)
Comment (brobeson) [reply cmake/Findlibzmq.cmake:2]: They're separate packages.

Comment (bnhhughes) [reply cmake/Findlibzmq.cmake:2]: 👍

File: readme.md (hunk 6 above)
Comment (brobeson) [reply readme.md]: Nope. Thanks for pointing this out.

File: readme.md (hunk 7 above)
Comment (brobeson) [reply readme.md]: I'll need to change that later. At this point, the version will be 20.05.1.

**Pull request merged**